lib/ansible/galaxy/collection/concrete_artifact_manager.py

```diff
@@ -338,6 +340,24 @@ def get_direct_collection_meta(self, collection):
         self._artifact_meta_cache[collection.src] = collection_meta
         return collection_meta
 
+    def get_requires_ansible(self, collection: Requirement) -> str | None:
+        """Extract requires_ansible from the on-disk collection artifact."""
+        b_artifact_path = self.get_artifact_path(collection)
+        if collection.is_url or collection.is_file:
+            runtime = _get_runtime_from_tar(b_artifact_path) or {}
+        elif collection.is_dir:
+            runtime = _get_runtime_from_dir(b_artifact_path) or {}
+        elif collection.is_virtual:
+            runtime = {}
```

Comment:
Would this be better?
```suggestion
            runtime = {'requires_ansible': '*'}
```

Reply:
I don't think so. I'd still need to assume the dictionary could be empty below. The file is optional, and can be empty. I also want to avoid conflating the collection version (which can be "*") with a collection's requires_ansible version (which cannot be "*", but can be None an empty string or absent to match any ansible version).